src/NServiceBus.Core/Serializers/SimpleJson/SimpleJson.g.cs

```diff
@@ -1067,7 +1067,24 @@ static bool SerializeObject(IJsonSerializerStrategy jsonSerializerStrategy, IEnu
                 if (stringKey != null)
                     SerializeString(stringKey, builder);
                 else
-                    SerializeString(key.ToString(), builder);
+                {
+//                    alternative approach:
+//                    jsonSerializerStrategy.TrySerializeNonPrimitiveObject(key, out var keyValue);
```

Comment:
this approach works too and can probably even handle `Uri` as keys (not tested though). It is quite a bit more expensive in case of "regular" key types like `int` etc, as it will go through the whole `TrySerializeNonPrimiteObject` path which also includes some reflection. This approach handles additional types besides DateTimes which are:
* Guids: they work already as proven by the test
* Enums: they don't work as the deserialization can't handle that
* Uris: current behavior not tested

Reply:
So what additional types do we gain from this?